### PR TITLE
Adding evaluation: VizWiz + TextVQA + fixing image loading bug

### PR DESCRIPTION
**Pull request by yonatanbitton**
1. Fixing the bug from [this issue](https://github.com/mlfoundations/open_flamingo/issues/148) by adding `image.load()` after opening the image.

2. Adding evaluation dataset VizWiz + TextVQA.
The method was to convert the datasets to the VQA format, jsons of questions and annotations, to make fewer edits in the evaluation code. 
Questions + Annotations are available [in this drive](https://drive.google.com/drive/folders/1fUsHqCCJ8oJ2u4EEIyeoNSvCYPo2uf36?usp=sharing). 
VizWiz Images: https://vizwiz.cs.colorado.edu/VizWiz_final/images/val.zip
TextVQA Images: the attached drive script `download_textvqa_images.py` allows to download
There are also bash files: `run_eval_vizwiz.sh` and `run_eval_textvqa.sh` to run the evaluations. 

Example addition for the [original eval script](https://github.com/mlfoundations/open_flamingo/blob/main/open_flamingo/scripts/run_eval.sh) (all described in the attached bash files):
```bash
VIZWIZ_IMG_PATH="${SERVER_DIR}/VizWiz/val"
VIZWIZ_ANNO_PATH="${SERVER_DIR}/VizWiz/Annotations/val_annotations_vqa_format.json"
VIZWIZ_QUESTION_PATH="${SERVER_DIR}/VizWiz/Annotations/val_questions_vqa_format.json"

python open_flamingo/eval/evaluate.py \
    --lm_path $LM_PATH \
    --lm_tokenizer_path $LM_TOKENIZER_PATH \
    ...
    ...
    --vizwiz_image_dir_path $VIZWIZ_IMG_PATH \
    --vizwiz_annotations_json_path $VIZWIZ_ANNO_PATH \
    --vizwiz_questions_json_path $VIZWIZ_QUESTION_PATH \
    --results_file $RESULTS_FILE \
    --eval_textvqa \
    ...
```

### CODE REVIEW

File: open_flamingo/eval/evaluate.py [hunk 1]
```diff
@@ -163,12 +175,53 @@
...
+    default=None,
+)
+
 ## Imagenet dataset
 parser.add_argument("--imagenet_root", type=str, default="/tmp")
 
 
 def main():
     args = parser.parse_args()
+    print(f"batch_size: {args.batch_size}")
```
Comment (anas-awadalla) [inline open_flamingo/eval/evaluate.py:224]: Can we remove this?

**Review by anas-awadalla — CHANGES_REQUESTED** [review]
Overall looks very clean. My only issue with this is more on the structure of the evaluation. I think we need to make sure to support validation and test evaluations. In the deepmind paper the former is done by querying examples and test samples from disjoint sets from the training split. The latter should query the examples from the training set but evaluate on the entire test (or sometimes validation) sets. The current implementation for vizwiz textvqa flicker seems to query the examples and test samples from the validation set? I think we should not do this as the numbers we get aren’t going to be accurate. Here is what I propose. Can we fix this PR to query from the training set rather than validation set? After that we should make another PR to add a flag to do either validation or test evaluations.
